runtime/src/enclave_rpc/client.rs

```diff
@@ -311,11 +363,30 @@ impl Controller {
             .write_message(types::Message::Close, &mut buffer)
             .map_err(|_| RpcClientError::Transport)?;
 
-        self.transport
-            .write_noise_session(self.session.id, buffer, String::new(), vec![node])
+        let request_id = self.increment_request_id();
+
+        let result = self
+            .transport
+            .write_noise_session(
+                request_id,
+                self.session.id,
+                buffer,
+                String::new(),
+                vec![node],
+            )
             .await
             .map_err(|_| RpcClientError::Transport)
-            .map(|(data, _)| data)
+            .map(|rsp| rsp.data);
+
+        let pf = if result.is_err() {
+            types::PeerFeedback::Failure
+        } else {
+            types::PeerFeedback::Success
+        };
+
+        let _ = self.transport.submit_peer_feedback(request_id, pf).await; // Ignore error.
```

Comment:
Maybe peer feedback should never be submitted for close messages, not sure :thinking:
For example, if call fails during connect, we will give a bad feedback, and another one when we gracefully close  the session.

Reply:
Yeah probably should not submit it for close as the error is ignored anyway. It serves mostly to notify the other side.